library/asn1parse.c

```diff
@@ -139,17 +139,18 @@ int mbedtls_asn1_get_bool( unsigned char **p,
     return( 0 );
 }
 
-int mbedtls_asn1_get_int( unsigned char **p,
-                  const unsigned char *end,
-                  int *val )
+static int asn1_get_tagged_int( unsigned char **p,
+                        const unsigned char *end,
+                        int tag, int *val )
 {
     int ret;
     size_t len;
 
-    if( ( ret = mbedtls_asn1_get_tag( p, end, &len, MBEDTLS_ASN1_INTEGER ) ) != 0 )
+    if( ( ret = mbedtls_asn1_get_tag( p, end, &len, tag ) ) != 0 )
         return( ret );
 
-    /* len==0 is malformed (0 must be represented as 020100). */
+    /* len==0 is malformed (0 must be represented as 020100 for INTEGER,
+       or 0A0100 for ENUMERATED tags */
```

Comment:
Style: comment continuation lines must start with `*` (aligned with the `*` of the opening `/*`), otherwise some of our sanity check scripts that don't perfectly understand C syntax can fail.

Reply:
Done